panic in graphite handler

against: 5a2b04229bce06e2b4074e8c0e07f53a3583f0f9

steps to repro:

1.
start influxd with the following graphite settings (everything else is stock)
```
# Configure the Graphite plugins.
[[graphite]] # 1 or more of these sections may be present.
enabled = true
protocol = "tcp" # Set to "tcp" or "udp"  
# address = "0.0.0.0" # If not set, is actually set to bind-address.port = 2003
# name-position = "last"
# name-separator = "-"
database = "stats"  # store graphite data in this database
```
2.
Run this:
```
echo "stats.counters.nerdsauna.counter.rate 0.000000 1426252648" | nc localhost 2003
```

=> crash

```

 8888888           .d888 888                   8888888b.  888888b.
   888            d88P"  888                   888  "Y88b 888  "88b
   888            888    888                   888    888 888  .88P
   888   88888b.  888888 888 888  888 888  888 888    888 8888888K.
   888   888 "88b 888    888 888  888  Y8bd8P' 888    888 888  "Y88b
   888   888  888 888    888 888  888   X88K   888    888 888    888
   888   888  888 888    888 Y88b 888 .d8""8b. 888  .d88P 888   d88P
 8888888 888  888 888    888  "Y88888 888  888 8888888P"  8888888P"

[http] 2015/03/14 06:36:46 127.0.0.1 - - [14/Mar/2015:06:36:46 +0000] GET /query?q=show%20series&db=stats HTTP/1.1 200 16 - curl/7.26.0 7c8aa3c2-ca14-11e4-a678-525400939b43
panic: unsupported value type: int64

goroutine 2532 [running]:
github.com/influxdb/influxdb.(*FieldCodec).EncodeFields(0xc2080955d0, 0xc20803dd10, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/database.go:798 +0xae4
github.com/influxdb/influxdb.func·031(0x0, 0x0)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/server.go:1598 +0x5fd
github.com/influxdb/influxdb.(*Server).WriteSeries(0xc2080b4000, 0xc20802bab0, 0x5, 0xc208094260, 0x7, 0xc208098880, 0x1, 0x1, 0xd7f4e0, 0x0, ...)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/server.go:1616 +0x91f
github.com/influxdb/influxdb/graphite.(*TCPServer).handleConnection(0xc2080b06c0, 0x7f53529afb00, 0xc208045780)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/graphite/graphite_tcp.go:83 +0x38c
created by github.com/influxdb/influxdb/graphite.func·001
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/graphite/graphite_tcp.go:54 +0x212

goroutine 1 [chan receive (nil chan)]:
main.execRun(0xc20800a010, 0x1, 0x1)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/cmd/influxd/main.go:121 +0x780
main.main()
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/cmd/influxd/main.go:70 +0x35b

goroutine 5 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 6 [chan receive]:
main.func·002()
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/cmd/influxd/main.go:199 +0xed
created by main.startProfiling
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/cmd/influxd/main.go:202 +0x2dc

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 7 [select]:
github.com/influxdb/influxdb/raft.(*Log).applier(0xc20802e140, 0xc208034480)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/raft/log.go:1137 +0x2f0
created by github.com/influxdb/influxdb/raft.(*Log).Open
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/raft/log.go:277 +0x561

goroutine 8 [select]:
github.com/influxdb/influxdb/raft.(*Log).leaderLoop(0xc20802e140, 0xc208034480, 0x0)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/raft/log.go:881 +0x625
github.com/influxdb/influxdb/raft.(*Log).stateLoop(0xc20802e140, 0xc208034480, 0x3, 0xc2080344e0)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/raft/log.go:666 +0x3ea
created by github.com/influxdb/influxdb/raft.(*Log).startStateLoop
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/raft/log.go:632 +0x98

goroutine 2537 [sleep]:
github.com/influxdb/influxdb/raft.func·001()
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/raft/clock.go:66 +0x33
created by github.com/influxdb/influxdb/raft.newClockChan
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/raft/clock.go:66 +0xfb

goroutine 11 [IO wait]:
net.(*pollDesc).Wait(0xc208010ca0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208010ca0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc208010c40, 0x0, 0x7f53529adc30, 0xc20802a860)
	/usr/local/go/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc2080444b0, 0xc208050de0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc2080444b0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:244 +0x4c
net/http.(*Server).Serve(0xc208034780, 0x7f53529ae608, 0xc2080444b0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1728 +0x92
net/http.Serve(0x7f53529ae608, 0xc2080444b0, 0x7f53529af948, 0xc2080ae500, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1606 +0xa1
main.func·003()
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/cmd/influxd/run.go:69 +0x80
created by main.Run
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/cmd/influxd/run.go:73 +0x8b0

goroutine 12 [select]:
github.com/influxdb/influxdb.(*Broker).continuousQueryLoop(0xc20803da70, 0xc208034660)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/broker.go:75 +0x10a
created by github.com/influxdb/influxdb.(*Broker).RunContinuousQueryLoop
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/broker.go:55 +0x78

goroutine 13 [select]:
github.com/influxdb/influxdb/messaging.(*Client).streamFromURL(0xc208034060, 0xc208010cb0, 0xc2080342a0, 0x0, 0x0)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/messaging/client.go:455 +0x829
github.com/influxdb/influxdb/messaging.(*Client).streamer(0xc208034060, 0xc2080342a0)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/messaging/client.go:407 +0x150
created by github.com/influxdb/influxdb/messaging.(*Client).Open
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/messaging/client.go:161 +0x27e

goroutine 14 [select]:
github.com/influxdb/influxdb.(*Server).processor(0xc2080b4000, 0x7f53529af828, 0xc208034060, 0xc208034360)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/server.go:2828 +0x2aa
created by github.com/influxdb/influxdb.(*Server).setClient
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/server.go:447 +0x132

goroutine 15 [select]:
github.com/influxdb/influxdb.func·009()
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/server.go:297 +0x109
created by github.com/influxdb/influxdb.(*Server).StartRetentionPolicyEnforcement
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/server.go:304 +0x1dd

goroutine 16 [select]:
github.com/influxdb/influxdb.func·011()
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/server.go:356 +0x116
created by github.com/influxdb/influxdb.(*Server).StartShardGroupsPreCreate
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/server.go:363 +0x1dd

goroutine 18 [IO wait]:
net.(*pollDesc).Wait(0xc208010f40, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208010f40, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc208010ee0, 0x0, 0x7f53529adc30, 0xc2080953f8)
	/usr/local/go/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc2080441d8, 0xc208020e00, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc2080441d8, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:244 +0x4c
net/http.(*Server).Serve(0xc208034ae0, 0x7f53529ae608, 0xc2080441d8, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1728 +0x92
net/http.Serve(0x7f53529ae608, 0xc2080441d8, 0x7f53529afa80, 0xc208095450, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1606 +0xa1
github.com/influxdb/influxdb/admin.(*Server).ListenAndServe(0xc2080b0690)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/admin/admin.go:38 +0x1d8
created by main.Run
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/cmd/influxd/run.go:127 +0x1617

goroutine 19 [IO wait]:
net.(*pollDesc).Wait(0xc208010680, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208010680, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc208010620, 0x0, 0x7f53529adc30, 0xc208094300)
	/usr/local/go/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc208044170, 0xc2080176e0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc208044170, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:244 +0x4c
github.com/influxdb/influxdb/graphite.func·001()
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/graphite/graphite_tcp.go:49 +0x54
created by github.com/influxdb/influxdb/graphite.(*TCPServer).ListenAndServe
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/graphite/graphite_tcp.go:56 +0x1ae

goroutine 20 [chan receive]:
github.com/influxdb/influxdb.(*Server).StartReportingLoop(0xc2080b4000, 0x8bacb0, 0x3, 0x62bafd696e7a121f, 0x0)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/server.go:3427 +0x97
created by main.Run
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/cmd/influxd/run.go:186 +0x201c

goroutine 2538 [sleep]:
github.com/influxdb/influxdb/raft.(*Log).heartbeater(0xc20802e140, 0x1, 0xc2080e8af0, 0xc2080ae4e0, 0xc208034600)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/raft/log.go:938 +0xe28
created by github.com/influxdb/influxdb/raft.(*Log).leaderLoop
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/raft/log.go:878 +0x20a

goroutine 30 [select]:
net/http.(*persistConn).readLoop(0xc2080a8160)
	/usr/local/go/src/net/http/transport.go:928 +0x9ce
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0xc9f

goroutine 32 [IO wait]:
net.(*pollDesc).Wait(0xc208011250, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208011250, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc2080111f0, 0xc20807c000, 0x1000, 0x1000, 0x0, 0x7f53529adc30, 0xc208095560)
	/usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc208044238, 0xc20807c000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:121 +0xdc
net/http.(*liveSwitchReader).Read(0xc208032868, 0xc20807c000, 0x1000, 0x1000, 0x2, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:214 +0xab
io.(*LimitedReader).Read(0xc2080aff00, 0xc20807c000, 0x1000, 0x1000, 0x2, 0x0, 0x0)
	/usr/local/go/src/io/io.go:408 +0xce
bufio.(*Reader).fill(0xc208035680)
	/usr/local/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadSlice(0xc208035680, 0xc2080a1b0a, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:295 +0x257
bufio.(*Reader).ReadLine(0xc208035680, 0x0, 0x0, 0x0, 0xc2080b0100, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:324 +0x62
net/textproto.(*Reader).readLineSlice(0xc20803ccc0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/textproto/reader.go:55 +0x9e
net/textproto.(*Reader).ReadLine(0xc20803ccc0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/textproto/reader.go:36 +0x4f
net/http.ReadRequest(0xc208035680, 0xc20801fa00, 0x0, 0x0)
	/usr/local/go/src/net/http/request.go:598 +0xcb
net/http.(*conn).readRequest(0xc208032820, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:586 +0x26f
net/http.(*conn).serve(0xc208032820)
	/usr/local/go/src/net/http/server.go:1162 +0x69e
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1751 +0x35e

goroutine 28 [IO wait]:
net.(*pollDesc).Wait(0xc208011100, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208011100, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc2080110a0, 0xc2080b6000, 0x1000, 0x1000, 0x0, 0x7f53529adc30, 0xc208095580)
	/usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc208044220, 0xc2080b6000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:121 +0xdc
net/http.noteEOFReader.Read(0x7f53529afb88, 0xc208044220, 0xc2080a8108, 0xc2080b6000, 0x1000, 0x1000, 0xc208098000, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc2080afe40, 0xc2080b6000, 0x1000, 0x1000, 0xc208012000, 0x0, 0x0)
	<autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc208035500)
	/usr/local/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Peek(0xc208035500, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xf0
net/http.(*persistConn).readLoop(0xc2080a80b0)
	/usr/local/go/src/net/http/transport.go:842 +0xa4
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0xc9f

goroutine 26 [IO wait]:
net.(*pollDesc).Wait(0xc208010d80, 0x77, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitWrite(0xc208010d80, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:93 +0x43
net.(*netFD).connect(0xc208010d20, 0x0, 0x0, 0x7f53529afb60, 0xc20800aa20, 0xecc95ce15, 0x15db98db, 0xd7f4e0, 0x0, 0x0)
	/usr/local/go/src/net/fd_unix.go:114 +0x24b
net.(*netFD).dial(0xc208010d20, 0x7f53529ae640, 0x0, 0x7f53529ae640, 0xc20803d530, 0xecc95ce15, 0x15db98db, 0xd7f4e0, 0x0, 0x0)
	/usr/local/go/src/net/sock_posix.go:139 +0x37a
net.socket(0x8cb2d0, 0x3, 0x2, 0x1, 0x0, 0xc20803d500, 0x7f53529ae640, 0x0, 0x7f53529ae640, 0xc20803d530, ...)
	/usr/local/go/src/net/sock_posix.go:91 +0x422
net.internetSocket(0x8cb2d0, 0x3, 0x7f53529ae640, 0x0, 0x7f53529ae640, 0xc20803d530, 0xecc95ce15, 0xc215db98db, 0xd7f4e0, 0x1, ...)
	/usr/local/go/src/net/ipsock_posix.go:137 +0x148
net.dialTCP(0x8cb2d0, 0x3, 0x0, 0xc20803d530, 0xecc95ce15, 0x15db98db, 0xd7f4e0, 0xc208052770, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:156 +0x125
net.dialSingle(0x8cb2d0, 0x3, 0x973a17, 0x13, 0x0, 0x0, 0x7f53529ae5d8, 0xc20803d530, 0xecc95ce15, 0x615db98db, ...)
	/usr/local/go/src/net/dial.go:235 +0x3f5
net.func·016(0xecc95ce15, 0xe15db98db, 0xd7f4e0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:163 +0x12c
net.dial(0x8cb2d0, 0x3, 0x7f53529ae5d8, 0xc20803d530, 0xc20809db08, 0xecc95ce15, 0xe15db98db, 0xd7f4e0, 0x0, 0x0, ...)
	/usr/local/go/src/net/fd_unix.go:40 +0x6f
net.(*Dialer).Dial(0xc2080301c0, 0x8cb2d0, 0x3, 0x973a17, 0x13, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:170 +0x4be
net.*Dialer.Dial·fm(0x8cb2d0, 0x3, 0x973a17, 0x13, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:38 +0x79
net/http.(*Transport).dial(0xc20805a120, 0x8cb2d0, 0x3, 0x973a17, 0x13, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:479 +0x84
net/http.(*Transport).dialConn(0xc20805a120, 0x0, 0x973a10, 0x4, 0x973a17, 0x13, 0xc2080b1a10, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:564 +0x1678
net/http.func·019()
	/usr/local/go/src/net/http/transport.go:520 +0x42
created by net/http.(*Transport).getConn
	/usr/local/go/src/net/http/transport.go:522 +0x335

goroutine 988 [chan receive]:
net/http.func·016()
	/usr/local/go/src/net/http/transport.go:507 +0x65
created by net/http.func·017
	/usr/local/go/src/net/http/transport.go:513 +0xba

goroutine 29 [select]:
net/http.(*persistConn).writeLoop(0xc2080a80b0)
	/usr/local/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0xcbc

goroutine 31 [select]:
net/http.(*persistConn).writeLoop(0xc2080a8160)
	/usr/local/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0xcbc

goroutine 33 [chan receive]:
github.com/influxdb/influxdb/messaging.(*Replica).WriteTo(0xc208030700, 0x7f53529afe40, 0xc208032000, 0x7f53529afe40, 0x0, 0x0)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/messaging/broker.go:1041 +0x6d3
github.com/influxdb/influxdb/messaging.(*Handler).stream(0xc208094fb0, 0x7f53529afda8, 0xc208032000, 0xc20801e5b0)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/messaging/handler.go:101 +0x1c5
github.com/influxdb/influxdb/messaging.(*Handler).ServeHTTP(0xc208094fb0, 0x7f53529afda8, 0xc208032000, 0xc20801e5b0)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/messaging/handler.go:54 +0x1c8
main.(*Handler).ServeHTTP(0xc2080ae500, 0x7f53529afda8, 0xc208032000, 0xc20801e5b0)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/cmd/influxd/handler.go:32 +0x248
net/http.serverHandler.ServeHTTP(0xc208034780, 0x7f53529afda8, 0xc208032000, 0xc20801e5b0)
	/usr/local/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc2080328c0)
	/usr/local/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1751 +0x35e

goroutine 34 [IO wait]:
net.(*pollDesc).Wait(0xc208011170, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208011170, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc208011110, 0xc2080b2000, 0x1000, 0x1000, 0x0, 0x7f53529adc30, 0xc2080955c0)
	/usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc208044230, 0xc2080b2000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:121 +0xdc
net/http.noteEOFReader.Read(0x7f53529afb88, 0xc208044230, 0xc2080a81b8, 0xc2080b2000, 0x1000, 0x1000, 0x41, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc2080afee0, 0xc2080b2000, 0x1000, 0x1000, 0x41, 0x0, 0x0)
	<autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc208035620)
	/usr/local/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadSlice(0xc208035620, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:295 +0x257
net/http/internal.readLine(0xc208035620, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/internal/chunked.go:110 +0x5a
net/http/internal.(*chunkedReader).beginChunk(0xc20803d230)
	/usr/local/go/src/net/http/internal/chunked.go:47 +0x46
net/http/internal.(*chunkedReader).Read(0xc20803d230, 0xc2080fb440, 0x16, 0x16, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/internal/chunked.go:77 +0xbb
net/http.(*body).readLocked(0xc2080310c0, 0xc2080fb440, 0x16, 0x16, 0xffffffff, 0x0, 0x0)
	/usr/local/go/src/net/http/transfer.go:584 +0x7a
net/http.(*body).Read(0xc2080310c0, 0xc2080fb440, 0x16, 0x16, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transfer.go:579 +0x115
net/http.(*bodyEOFSignal).Read(0xc208031140, 0xc2080fb440, 0x16, 0x16, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1193 +0x285
io.ReadAtLeast(0x7f53529b0040, 0xc208031140, 0xc2080fb440, 0x16, 0x16, 0x16, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:298 +0xf1
io.ReadFull(0x7f53529b0040, 0xc208031140, 0xc2080fb440, 0x16, 0x16, 0x25, 0x0, 0x0)
	/usr/local/go/src/io/io.go:316 +0x6d
github.com/influxdb/influxdb/messaging.(*MessageDecoder).Decode(0xc208095250, 0xc2080b1aa0, 0x0, 0x0)
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/messaging/broker.go:1161 +0xa4
github.com/influxdb/influxdb/messaging.func·012()
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/messaging/client.go:442 +0x113
created by github.com/influxdb/influxdb/messaging.(*Client).streamFromURL
	/home/sampti/Code/go/src/github.com/influxdb/influxdb/messaging/client.go:452 +0x64a

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,7 @@
 
 ### Bugfixes
 - [#1942](https://github.com/influxdb/influxdb/pull/1942): Sort wildcard names.
+- [#1957](https://github.com/influxdb/influxdb/pull/1957): Graphite numbers are always float64.
 
 ## v0.9.0-rc11 [2015-03-13]
 

--- a/cmd/influxd/server_integration_test.go
+++ b/cmd/influxd/server_integration_test.go
@@ -1039,7 +1039,7 @@ func Test_ServerSingleGraphiteIntegration(t *testing.T) {
 	}
 }
 
-func Test_ServerSingleGraphiteIntegration_NoDatabase(t *testing.T) {
+func Test_ServerSingleGraphiteIntegration_ZeroDataPoint(t *testing.T) {
 	if testing.Short() {
 		t.Skip()
 	}
@@ -1051,7 +1051,58 @@ func Test_ServerSingleGraphiteIntegration_NoDatabase(t *testing.T) {
 	c := main.NewConfig()
 	g := main.Graphite{
 		Enabled:  true,
+		Database: "graphite",
+		Protocol: "TCP",
 		Port:     2103,
+	}
+	c.Graphites = append(c.Graphites, g)
+
+	t.Logf("Graphite Connection String: %s\n", g.ConnectionString(c.BindAddress))
+	nodes := createCombinedNodeCluster(t, testName, dir, nNodes, basePort, c)
+
+	createDatabase(t, testName, nodes, "graphite")
+	createRetentionPolicy(t, testName, nodes, "graphite", "raw")
+
+	// Connect to the graphite endpoint we just spun up
+	conn, err := net.Dial("tcp", g.ConnectionString(c.BindAddress))
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+
+	t.Log("Writing data")
+	data := []byte(`cpu 0.000 `)
+	data = append(data, []byte(fmt.Sprintf("%d", now.UnixNano()/1000000))...)
+	data = append(data, '\n')
+	_, err = conn.Write(data)
+	conn.Close()
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+
+	expected := fmt.Sprintf(`{"results":[{"series":[{"name":"cpu","columns":["time","cpu"],"values":[["%s",0]]}]}]}`, now.Format(time.RFC3339Nano))
+
+	// query and wait for results
+	got, ok := queryAndWait(t, nodes, "graphite", `select * from "graphite"."raw".cpu`, expected, 2*time.Second)
+	if !ok {
+		t.Errorf(`Test "%s" failed, expected: %s, got: %s`, testName, expected, got)
+	}
+}
+
+func Test_ServerSingleGraphiteIntegration_NoDatabase(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	nNodes := 1
+	basePort := 8590
+	testName := "graphite integration"
+	dir := tempfile()
+	now := time.Now().UTC().Round(time.Millisecond)
+	c := main.NewConfig()
+	g := main.Graphite{
+		Enabled:  true,
+		Port:     2203,
 		Protocol: "TCP",
 	}
 	c.Graphites = append(c.Graphites, g)

--- a/graphite/graphite.go
+++ b/graphite/graphite.go
@@ -86,12 +86,7 @@ func (p *Parser) Parse(line string) (influxdb.Point, error) {
 	}
 
 	fieldValues := make(map[string]interface{})
-	// Determine if value is a float or an int.
-	if i := int64(v); float64(i) == v {
-		fieldValues[name] = int64(v)
-	} else {
-		fieldValues[name] = v
-	}
+	fieldValues[name] = v
 
 	// Parse timestamp.
 	unixTime, err := strconv.ParseInt(fields[2], 10, 64)

--- a/graphite/graphite_test.go
+++ b/graphite/graphite_test.go
@@ -61,9 +61,7 @@ func Test_DecodeMetric(t *testing.T) {
 		line                string
 		name                string
 		tags                map[string]string
-		isInt               bool
-		iv                  int64
-		fv                  float64
+		value               float64
 		timestamp           time.Time
 		position, separator string
 		err                 string
@@ -73,8 +71,7 @@ func Test_DecodeMetric(t *testing.T) {
 			line:      `cpu.foo.bar 50 ` + strTime,
 			name:      "cpu",
 			tags:      map[string]string{"foo": "bar"},
-			isInt:     true,
-			iv:        50,
+			value:     50,
 			timestamp: testTime,
 		},
 		{
@@ -83,8 +80,7 @@ func Test_DecodeMetric(t *testing.T) {
 			line:      `cpu.foo.bar 50 ` + strTime,
 			name:      "cpu",
 			tags:      map[string]string{"foo": "bar"},
-			isInt:     true,
-			iv:        50,
+			value:     50,
 			timestamp: testTime,
 		},
 		{
@@ -93,8 +89,7 @@ func Test_DecodeMetric(t *testing.T) {
 			line:      `foo.bar.cpu 50 ` + strTime,
 			name:      "cpu",
 			tags:      map[string]string{"foo": "bar"},
-			isInt:     true,
-			iv:        50,
+			value:     50,
 			timestamp: testTime,
 		},
 		{
@@ -103,8 +98,7 @@ func Test_DecodeMetric(t *testing.T) {
 			line:      `cpu 50 ` + strTime,
 			name:      "cpu",
 			tags:      map[string]string{},
-			isInt:     true,
-			iv:        50,
+			value:     50,
 			timestamp: testTime,
 		},
 		{
@@ -113,8 +107,7 @@ func Test_DecodeMetric(t *testing.T) {
 			line:      `cpu 50 ` + strTime,
 			name:      "cpu",
 			tags:      map[string]string{},
-			isInt:     true,
-			iv:        50,
+			value:     50,
 			timestamp: testTime,
 		},
 		{
@@ -122,8 +115,7 @@ func Test_DecodeMetric(t *testing.T) {
 			line:      `cpu.foo.bar 50 ` + strTime,
 			name:      "cpu",
 			tags:      map[string]string{"foo": "bar"},
-			isInt:     true,
-			iv:        50,
+			value:     50,
 			timestamp: testTime,
 		},
 		{
@@ -132,8 +124,7 @@ func Test_DecodeMetric(t *testing.T) {
 			line:      `cpu.foo.bar 50 ` + strTime,
 			name:      "cpu",
 			tags:      map[string]string{"foo": "bar"},
-			isInt:     true,
-			iv:        50,
+			value:     50,
 			timestamp: testTime,
 		},
 		{
@@ -142,8 +133,7 @@ func Test_DecodeMetric(t *testing.T) {
 			line:      `cpu-foo-bar 50 ` + strTime,
 			name:      "cpu",
 			tags:      map[string]string{"foo": "bar"},
-			isInt:     true,
-			iv:        50,
+			value:     50,
 			timestamp: testTime,
 		},
 		{
@@ -152,8 +142,7 @@ func Test_DecodeMetric(t *testing.T) {
 			line:      `cpuboofooboobar 50 ` + strTime,
 			name:      "cpu",
 			tags:      map[string]string{"foo": "bar"},
-			isInt:     true,
-			iv:        50,
+			value:     50,
 			timestamp: testTime,
 		},
 
@@ -162,16 +151,14 @@ func Test_DecodeMetric(t *testing.T) {
 			line:      `cpu.foo.bar 50 ` + strTime,
 			name:      "cpu",
 			tags:      map[string]string{"foo": "bar"},
-			isInt:     true,
-			iv:        50,
+			value:     50,
 			timestamp: testTime,
 		},
 		{
 			test:      "metric only with float value",
 			line:      `cpu 50.554 ` + strTime,
 			name:      "cpu",
-			isInt:     false,
-			fv:        50.554,
+			value:     50.554,
 			timestamp: testTime,
 		},
 		{
@@ -224,16 +211,9 @@ func Test_DecodeMetric(t *testing.T) {
 		if len(point.Tags) != len(test.tags) {
 			t.Fatalf("tags len mismatch.  expected %d, got %d", len(test.tags), len(point.Tags))
 		}
-		if test.isInt {
-			i := point.Fields[point.Name].(int64)
-			if i != test.iv {
-				t.Fatalf("integerValue value mismatch.  expected %v, got %v", test.iv, point.Fields[point.Name])
-			}
-		} else {
-			f := point.Fields[point.Name].(float64)
-			if point.Fields[point.Name] != f {
-				t.Fatalf("floatValue value mismatch.  expected %v, got %v", test.fv, f)
-			}
+		f := point.Fields[point.Name].(float64)
+		if point.Fields[point.Name] != f {
+			t.Fatalf("floatValue value mismatch.  expected %v, got %v", test.value, f)
 		}
 		if point.Timestamp.UnixNano()/1000000 != test.timestamp.UnixNano()/1000000 {
 			t.Fatalf("timestamp value mismatch.  expected %v, got %v", test.timestamp.UnixNano(), point.Timestamp.UnixNano())

